### PR TITLE
feat(editor): Select credentials in template setup if theres only one

### PR DESCRIPTION
**Pull request by tomi**
Automatically select credentials in the template credential setup if there's only one available.

NOTE! This feature is still behind a feature flag. To test, set:

```js
localStorage.setItem('template-credentials-setup', 'true')
```

https://github.com/n8n-io/n8n/assets/10324676/edc1f586-214f-4c37-b7ec-dd1786433dc1

### CODE REVIEW

File: packages/editor-ui/src/views/SetupWorkflowFromTemplateView/setupTemplate.store.ts [hunk 1]
```diff
@@ -232,6 +232,20 @@ export const useSetupTemplateStore = defineStore('setupTemplate', () => {
...
 
+	/**
+	 * Selects initial credentials for the template. Credentials
+	 * need to be loaded before this.
+	 */
+	const setInitialCredentialSelection = () => {
+		for (const credUsage of credentialUsages.value) {
+			const availableCreds = credentialsStore.getCredentialsByType(credUsage.credentialType);
+
+			if (availableCreds.length === 1) {
```
Comment (mutdmour) [inline packages/editor-ui/src/views/SetupWorkflowFromTemplateView/setupTemplate.store.ts:243]: should we avoid doing this for generic http credentials? I think we have some logic around that for automatically setting creds in NDV

Comment (tomi) [reply packages/editor-ui/src/views/SetupWorkflowFromTemplateView/setupTemplate.store.ts:257]: I couldn't find any sophisticated logic how those would be ignore. I added a simple exclude list for those credential types. Does that work @mutdmour ?

Comment (mutdmour) [reply packages/editor-ui/src/views/SetupWorkflowFromTemplateView/setupTemplate.store.ts:257]: Yup.. Looks good to me.. Sorry I don't recall specifics of that, but I find it really annoying usually when it auto selects these cred types.

**Review by mutdmour — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.20.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.20.0)

**Pull request merged**